libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAILTNPublisher.m

```diff
@@ -65,21 +66,14 @@ + (BOOL) publishLogLines:(NSDictionary *) bodyDictionary {
     [request setCustomRequestBodyData:postData contentType:@"application/json"];
     [request setHeaderValue:@"gzip" forHeaderName:@"Content-Encoding"];
     [request setHeaderValue:[NSString stringWithFormat:@"%lu", (unsigned long)[postData length]] forHeaderName:@"Content-Length"];
-    __block BOOL finished = NO;
-    __block BOOL success = NO;
     [[SFRestAPI sharedInstance] sendRESTRequest:request failBlock:^(NSError *e) {
-        finished = YES;
         if (e) {
             [SFLogger log:[self class] level:SFLogLevelError format:@"Upload failed %ld %@", (long)[e code], [e localizedDescription]];
         }
+        publishCompleteBlock(NO);
     } completeBlock:^(id response) {
-        finished = YES;
-        success = YES;
+        publishCompleteBlock(YES);
     }];
-    while(!finished) {
```

Comment:
This is primarily what I wanted to get rid of. This run loop can potentially block a thread forever. I was using it just to pass synchronous result back, but I refactored this to call the completion block instead.